### PR TITLE
Add client structure

### PR DESCRIPTION
**Pull request by willsb**
Squash whenever

### CODE REVIEW

File: Toggl.Ultrawave/Clients/Interfaces/IClientsClient.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Toggl.Ultrawave.Clients
+{
+	public interface IClientsClient
```
Comment (amulware) [inline Toggl.Ultrawave/Clients/Interfaces/IClientsClient.cs]: This is somewhat confusing naming. Any way we can clear that up?

Comment (willsb) [inline Toggl.Ultrawave/Clients/Interfaces/IClientsClient.cs]: Somewhat? This is very confusing 😄 

Though, by definition, this is exactly what we need if we follow the lego naming pattern (and there's no better suffix than `Client`).

In real use, it won't hurt much, though:

```C#
var api = new TogglApiClient();
api.Clients.GetAll();
```

File: Toggl.Ultrawave/Clients/ClientsClient.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Toggl.Ultrawave.Clients
+{
+    public class ClientsClient : IClientsClient
```
Comment (amulware) [inline Toggl.Ultrawave/Clients/ClientsClient.cs]: Why is this public?

Comment (amulware) [inline Toggl.Ultrawave/Clients/ClientsClient.cs]: (same below)

Comment (willsb) [inline Toggl.Ultrawave/Clients/ClientsClient.cs]: You are right. They should be both `internal` and `sealed`

Comment (amulware) [inline Toggl.Ultrawave/Clients/ClientsClient.cs]: Should we make it policy to seal all classes that arent specifically meant to be inherited from?

Comment (willsb) [inline Toggl.Ultrawave/Clients/ClientsClient.cs]: Ideally yes. Makes our intention explicit.

Comment (amulware) [inline Toggl.Ultrawave/Clients/ClientsClient.cs]: Agreed. See #16

File: Toggl.Ultrawave/ApiEnvironment.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Toggl.Ultrawave
+{
+    public enum ApiEnvironment
+	{
```
Comment (amulware) [inline Toggl.Ultrawave/ApiEnvironment.cs]: Formatting.

**Pull request merged**